presto-parser/src/test/java/com/facebook/presto/sql/parser/TestSqlParser.java

```diff
@@ -1645,19 +1651,60 @@ public void testExists()
     }
 
     @Test
-    public void testShowColumnStats()
+    public void testShowStats()
     {
         final String[] tableNames = {"t", "s.t", "c.s.t"};
 
         for (String fullName : tableNames) {
             QualifiedName qualifiedName = QualifiedName.of(Arrays.asList(fullName.split("\\.")));
             assertStatement(format("SHOW STATS %s", qualifiedName), new ShowStats(qualifiedName));
-            assertStatement(format("SHOW STATS FROM %s", qualifiedName), new ShowStats(qualifiedName));
-            assertStatement(format("SHOW STATS IN %s", qualifiedName), new ShowStats(qualifiedName));
+            assertStatement(format("SHOW STATS FOR %s", qualifiedName), new ShowStats(qualifiedName));
+            assertStatement(format("SHOW STATS ON %s", qualifiedName), new ShowStats(qualifiedName));
         }
     }
 
     @Test
+    public void testShowStatsParametrized()
+    {
+        final String[] tableNames = {"t", "s.t", "c.s.t"};
+
+        for (String fullName : tableNames) {
```

Comment:
It covers cases we managed to do wrong when we first started implementing that with @ArturGajowy